source/simulator/solver_schemes.cc

```diff
@@ -440,7 +441,7 @@ namespace aspect
     // don't need to force assembly of the matrix.
     if (stokes_matrix_depends_on_solution()
         ||
-        (nonlinear_iteration == 0 && boundary_velocity_manager.get_active_boundary_velocity_conditions().size() > 0))
+        (nonlinear_iteration == 0 ))
```

Comment:
Is this an accident? I think that the original condition should have had an `||` instead of `&&` -- that strikes me as a bug in the original code. Or do I misunderstand the situation?